src/EventSubscriber/PlayerSubscriber.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FrankProjects\UltimateWarfare\EventSubscriber;
+
+use FrankProjects\UltimateWarfare\Entity\User;
+use FrankProjects\UltimateWarfare\Service\GameEngine;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+final class PlayerSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var GameEngine
+     */
+    private $gameEngine;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * PlayerSubscriber constructor.
+     *
+     * @param GameEngine $gameEngine
+     * @param SessionInterface $session
+     * @param TokenStorageInterface $tokenStorage
+     * @param RouterInterface $router
+     */
+    public function __construct(
+        GameEngine $gameEngine,
+        SessionInterface $session,
+        TokenStorageInterface $tokenStorage,
+        RouterInterface $router
+    ) {
+        $this->gameEngine = $gameEngine;
+        $this->session = $session;
+        $this->tokenStorage = $tokenStorage;
+        $this->router = $router;
+    }
+
+    /**
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event): void
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $user = $this->getUser();
+        if ($user === null) {
+            return;
+        }
+
+        if ($user->getActive() === false) {
+            $this->checkBannedAndRedirect($event);
+        } else {
+            $this->runGameEngine($user);
```

Comment:
Although i am not sure as to _how_ the game engine advances state calling this here will indeed ensure the call is made during every request.

Reply:
Thanks!